pkg/api/operators/apis/experiment/v1alpha2/types.go

```diff
@@ -0,0 +1,194 @@
+
+	// Should we include more metadata about Trials here?
+	// E.g. Number of Trials completed/failed/pending
+}
+
+// +k8s:deepcopy-gen=true
+// ExperimentCondition describes the state of the experiment at a certain point.
+type ExperimentCondition struct {
+	// Type of experiment condition.
+	Type ExperimentConditionType `json:"type"`
+
+	// Status of the condition, one of True, False, Unknown.
+	Status v1.ConditionStatus `json:"status"`
+
+	// The reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+
+	// A human readable message indicating details about the transition.
+	Message string `json:"message,omitempty"`
+
+	// The last time this condition was updated.
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty"`
+
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+}
+
+// ExperimentConditionType defines the state of an Experiment.
+type ExperimentConditionType string
+
+const (
+	ExperimentCreated    ExperimentConditionType = "Created"
+	ExperimentRunning    ExperimentConditionType = "Running"
+	ExperimentRestarting ExperimentConditionType = "Restarting"
+	ExperimentSucceeded  ExperimentConditionType = "Succeeded"
+	ExperimentFailed     ExperimentConditionType = "Failed"
+)
+
+type ParameterSpec struct {
+	Name          string        `json:"name,omitempty"`
+	ParameterType ParameterType `json:"parameterType,omitempty"`
+	FeasibleSpace FeasibleSpace `json:"feasibleSpace,omitempty"`
+}
+
+type ParameterType string
+
+const (
+	ParameterTypeUnknown     ParameterType = "unknown"
+	ParameterTypeDouble      ParameterType = "double"
+	ParameterTypeInt         ParameterType = "int"
+	ParameterTypeDiscrete    ParameterType = "discrete"
+	ParameterTypeCategorical ParameterType = "categorical"
+)
+
+type FeasibleSpace struct {
+	Max  string   `json:"max,omitempty"`
+	Min  string   `json:"min,omitempty"`
+	List []string `json:"list,omitempty"`
+	Step string   `json:"step,omitempty"`
+}
+
+type ObjectiveSpec struct {
+	Type                ObjectiveType `json:"type,omitempty"`
+	Goal                float64       `json:"goal,omitempty"`
+	ObjectiveMetricName string        `json:"objectiveMetricName,omitempty"`
+	// This can be empty if we only care about the objective metric.
+	// Note: If we adopt a push instead of pull mechanism, this can be omitted completely.
+	AdditionalMetricsNames []string `json:"additionalMetricsNames,omitempty"`
+}
+
+type ObjectiveType string
+
+const (
+	ObjectiveTypeUnknown  ObjectiveType = ""
+	ObjectiveTypeMinimize ObjectiveType = "minimize"
+	ObjectiveTypeMaximize ObjectiveType = "maximize"
+)
+
+type AlgorithmSpec struct {
+	Suggestion    *SuggestionSpec    `json:"suggestion,omitempty"`
+	EarlyStopping *EarlyStoppingSpec `json:"earlyStopping,omitempty"`
+}
+
+type SuggestionSpec struct {
+	Name       string                   `json:"name,omitempty"`
```

Comment:
This refers to algorithm name right?

Reply:
I'm also unsure what Suggestion represents here

Reply:
Sorry - this is the suggestion algorithm name. @johnugeorge did we decide to move this to the AlgorithmSpec?

Reply:
Do you mean moving SuggestionSpec to AlgorithmSpec and remove EarlyStopping?
IMO, we should keep both of Suggestion and EarlyStopping.
To avoid confusing, we can rename the field `AlgorithmName`.

Reply:
I meant moving the field `Name` to AlgorithmSpec, keep the rest the same. I can also rename it as `AlgorithmName` to avoid confusion.